
<launch>
    <arg name="name"                                default="downsampler"/>
    <arg name="manager"                             default="manager"/>
    <arg name="input_cloud_topic"                   default="camera/depth/points"/>
    <arg name="downsampled_points_topic"            default="camera/depth/downsampled_points"/>

    <arg name="min_range"                           default="0.1"/>
    <arg name="max_range"                           default="10.0"/>
    <arg name="leaf_size"                           default="0.01"/>
    <arg name="filter_radius"                       default="0.03"/>
    <arg name="min_points_threshold"                default="3"/>
    <arg name="rate"                                default="30"/>

    <arg name="sensor_frame"                        default="camera_depth_optical_frame"/>
    <arg name="sensor_frame_overwrite"              default=""/>
    <arg name="cut_off_distance"                    default="1.2"/>

    <node pkg="nodelet" type="nodelet" name="$(arg name)" args="load downsampler/downsampler $(arg manager)">
        <remap from="input_cloud"                   to="$(arg input_cloud_topic)"/>
        <remap from="downsampled_points"            to="$(arg downsampled_points_topic)"/>

        <param name="min_range"                        type="double"         value="$(arg min_range)"/>
        <param name="max_range"                        type="double"         value="$(arg max_range)"/>
        <param name="leaf_size"                        type="double"         value="$(arg leaf_size)"/>
        <param name="filter_radius"                    type="double"         value="$(arg filter_radius)"/>
        <param name="min_points_threshold"             type="int"            value="$(arg min_points_threshold)"/>
        <param name="rate"                             type="double"         value="$(arg rate)"/>

        <param name="sensor_frame"                     type="string"         value="$(arg sensor_frame)"/>
        <param name="sensor_frame_overwrite"           type="string"         value="$(arg sensor_frame_overwrite)"/>
        <param name="cut_off_distance"                 type="double"         value="$(arg cut_off_distance)"/>
    </node>
</launch>
